cli/src/commands/git/clone.rs

```diff
@@ -60,6 +61,11 @@ pub struct GitCloneArgs {
     /// Whether or not to colocate the Jujutsu repo with the git repo
     #[arg(long)]
     colocate: bool,
+    /// Create a shallow clone of the given depth
+    // We accept a depth of 0 for parity with git's --depth option.
+    // The option is here to not render a default = 0 in the help message.
```

Comment:
nit: I don't think we need to copy the exact git CLI behavior. If `--depth 0` doesn't make sense, I would make it an error.